File: 品味Java的21種設計模式/dp_cpp/abstractfactory/example5/RdbDAOFactory.cpp
#include "RdbDAOFactory.h"

namespace cn {
namespace javass {
namespace dp {
namespace abstractfactory {
namespace example5 {

OrderDetailDAO *RdbDAOFactory::createOrderDetailDAO() {
    return new RdbDetailDAOImpl();
}

OrderMainDAO *RdbDAOFactory::createOrderMainDAO() {
    return new RdbMainDAOImpl();
}
}
}
}
}
}

File: 品味Java的21種設計模式/dp_java/abstractfactory/example5/RdbDAOFactory.java
package cn.javass.dp.abstractfactory.example5;

public class RdbDAOFactory extends DAOFactory {
    public OrderDetailDAO createOrderDetailDAO() {
        return new RdbDetailDAOImpl();
    }
    public OrderMainDAO createOrderMainDAO() {
        return new RdbMainDAOImpl();
    }
}
